src/pages/studyView/menu/AutosubmitToggle.tsx

```diff
@@ -4,22 +4,21 @@ import classNames from 'classnames';
 import styles from 'pages/studyView/styles.module.scss';
 import Tooltip from 'rc-tooltip';
 
-import { StudyViewPageStore } from 'pages/studyView/StudyViewPageStore';
-import { StudyViewContext } from 'pages/studyView/StudyViewContext';
+import autobind from 'autobind-decorator';
 export type IAutosubmitToggleProps = {
-    store: StudyViewPageStore;
+    hesitateModeOn: boolean;
+    onChange: (manual: boolean) => void;
 };
 
-export const STUDY_VIEW_FILTER_AUTOSUBMIT = 'study_view_filter_autosubmit';
-
 @observer
 export class AutosubmitToggle extends React.Component<
     IAutosubmitToggleProps,
     {}
 > {
-    private updateHesitateMode(hesitateMode: boolean) {
-        localStorage.setItem(STUDY_VIEW_FILTER_AUTOSUBMIT, '' + hesitateMode);
-        this.context.store.hesitateUpdate = hesitateMode;
+    @autobind
```

Comment:
@BasLee the idea of this refactor is that we should not overuse React context.  It's better to pass in a callback when the parent has direct access to the necessary things (the store) and there's only one (or a few) parents.

Reply:
check!